plugins/woocommerce/src/Blocks/BlockTypes/ProductCollection/Controller.php

```diff
@@ -112,13 +136,40 @@ public function block_hook_fallback_store_notice( $hooked_blocks, $position, $an
 	 *                                                             or pattern that the anchor block belongs to.
 	 * @return array|null
 	 */
-	public function augment_hooked_store_notices_block( $parsed_hooked_block, $hooked_block_type, $relative_position, $parsed_anchor_block, $context ) {
+	public function augment_hooked_store_notices_block( $parsed_hooked_block, $hooked_block_type, $relative_position, $parsed_anchor_block, $context ) {		
+		// Don't hook the block if there is not a product collection block in the content.
+		if ( $context instanceof \WP_Post && ! str_contains( $context->post_content, 'woocommerce/product-collection' ) ) {
+			return null;
+		}
+
+		
 		// Do not default to wide, as it will break the layout for this fallback block.
 		$parsed_hooked_block['attrs']['align'] = '';
 
 		return $parsed_hooked_block;
 	}
 
+	public function block_hooks_support_post_content() {
```

Comment:
We can probably simplify this code, looking [here](https://github.com/WordPress/gutenberg/pull/67272/files#diff-76236aad75203b8bb9c5c1bb8c6c6bdd9751d9226f624c4e89327af5c11cb0b4R155) and [here](https://github.com/WordPress/wordpress-develop/pull/7889/files#diff-4112a9a0bb18713bc2fc7868ceffe4403d765722f605e42b313ef2cdc218177fR195) we could in theory do something like:

```php
if ( has_filter('the_content', 'gutenberg_apply_block_hooks_to_post_content') || has_filter('the_content', 'apply_block_hooks_to_content') ) {
	// ...
}
```

Reply:
I think it might be better being moved into [wc-conditional-functions.php](https://github.com/woocommerce/woocommerce/blob/trunk/plugins/woocommerce/includes/wc-conditional-functions.php) too

Reply:
Oh nice, great suggestions thank you